.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,30 @@
+# Adapted from https://github.com/marketplace/actions/deploy-pr-preview.
+name: Deploy PR Preview
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+
+concurrency: preview-${{ github.ref }}
+
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+        # Must override default vite build base so that assets can still be accessed when not placed in the root directory of gh-pages branch.
+      - name: Install and Build
+        run: npm ci && npx tsc && npx vite build --base=https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-${PR_NUMBER}/
```

Comment:
Isn't the `tsc` step unneeded?  I thought vite build would do the heavy lifting there.

Reply:
Hmm, you're right, vite does do transpilation for you! 

There's a note in the documentation that vite does not do typechecking, and they recommend that you either run tsc separately (which we're doing in static checks) or add an additional tsc step before the vite build command: https://vitejs.dev/guide/features.html#transpile-only